src/Nancy/Json/Simple/NancySerializationStrategy.cs

```diff
@@ -0,0 +1,231 @@
+    using System.Globalization;
+    using System.Linq;
+    using System.Reflection;
+    using Nancy.Extensions;
+
+    /// <summary>
+    /// Nancy serialization stategy for SimpleJson
+    /// </summary>
+    public class NancySerializationStrategy : PocoJsonSerializerStrategy
+    {
+        private readonly bool retainCasing;
+        private readonly List<JavaScriptConverter> converters = new List<JavaScriptConverter>();
+        private readonly List<JavaScriptPrimitiveConverter> primitiveConverters = new List<JavaScriptPrimitiveConverter>();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NancySerializationStrategy"/> class.
+        /// </summary>
+        public NancySerializationStrategy() : this(false)
+        {
+
+        }
+
+        /// <summary>
+        ///  Initializes a new instance of the <see cref="NancySerializationStrategy"/> class.
+        /// </summary>
+        /// <param name="retainCasing">Retain C# casing of objects when serialized</param>
+        public NancySerializationStrategy(bool retainCasing)
+        {
+            this.retainCasing = retainCasing;
+        }
+
+        /// <summary>
+        /// Register custom <see cref="JavaScriptConverter"/> converters
+        /// </summary>
+        /// <param name="javaScriptConverters">An array of <see cref="JavaScriptConverter"/></param>
+        public void RegisterConverters(IEnumerable<JavaScriptConverter> javaScriptConverters)
+        {
+            this.converters.AddRange(javaScriptConverters);
+        }
+
+        /// <summary>
+        /// Register custom <see cref="JavaScriptPrimitiveConverter"/>
+        /// </summary>
+        /// <param name="javaScriptPrimitiveConverters">An array of <see cref="JavaScriptPrimitiveConverter"/></param>
+        public void RegisterConverters(IEnumerable<JavaScriptPrimitiveConverter> javaScriptPrimitiveConverters)
+        {
+            this.primitiveConverters.AddRange(javaScriptPrimitiveConverters);
+        }
+
+        /// <summary>
+        /// Formats a property name to a JSON field name
+        /// </summary>
+        /// <param name="clrPropertyName">The property name to format</param>
+        /// <returns></returns>
+        protected override string MapClrMemberNameToJsonFieldName(string clrPropertyName)
+        {
+            return this.retainCasing
+                ? base.MapClrMemberNameToJsonFieldName(clrPropertyName)
+                : clrPropertyName.ToCamelCase();
+        }
+
+        /// <summary>
+        /// Deserialize an object
+        /// </summary>
+        /// <param name="value">The object to deserialize</param>
+        /// <param name="type">The type of object to deserialize</param>
+        /// <returns></returns>
+        public override object DeserializeObject(object value, Type type)
+        {
+            if (type.IsEnum || (ReflectionUtils.IsNullableType(type) && Nullable.GetUnderlyingType(type).IsEnum))

```

Comment:
Both of these `IsEnum` calls needs to happen on `type.GetTypeInfo().IsEnum` or it will not work on CoreCLR